core/src/main/java/nl/nn/adapterframework/util/flow/FlowDiagramManager.java

```diff
@@ -61,97 +57,59 @@ public class FlowDiagramManager implements ApplicationContextAware, Initializing
 	private static final AppConstants APP_CONSTANTS = AppConstants.getInstance();
 	private File adapterFlowDir = new File(APP_CONSTANTS.getResolvedProperty("flow.adapter.dir"));
 	private File configFlowDir = new File(APP_CONSTANTS.getResolvedProperty("flow.config.dir"));
-	private ApplicationContext applicationContext;
-
-	private static final String ADAPTER2DOT_XSLT = "/xml/xsl/adapter2dot.xsl";
-	private static final String CONFIGURATION2DOT_XSLT = "/xml/xsl/configuration2dot.xsl";
-	private static final String NO_IMAGE_AVAILABLE = "/no_image_available.svg";
+	private @Setter ApplicationContext applicationContext;
 
-	private TransformerPool transformerPoolAdapter;
-	private TransformerPool transformerPoolConfig;
-	private URL noImageAvailable;
-	private String fileExtension = null;
-
-	/**
-	 * Optional IFlowGenerator. If non present the FlowDiagramManager should still be 
-	 * able to generate dot files and return the `noImageAvailable` image.
-	 */
-	private ThreadLocal<SoftReference<IFlowGenerator>> generators = new ThreadLocal<SoftReference<IFlowGenerator>>();
+	private IFlowGenerator flowGenerator;
 
 	@Override
 	public void afterPropertiesSet() throws Exception {
-		Resource xsltSourceConfig = Resource.getResource(ADAPTER2DOT_XSLT);
-		transformerPoolAdapter = TransformerPool.getInstance(xsltSourceConfig, 2);
-
-		Resource xsltSourceIbis = Resource.getResource(CONFIGURATION2DOT_XSLT);
-		transformerPoolConfig = TransformerPool.getInstance(xsltSourceIbis, 2);
+		if(applicationContext == null) {
+			throw new IllegalStateException("ApplicationContext has not been autowired, cannot instantiate IFlowDiagram");
+		}
 
-		IFlowGenerator generator = getFlowGenerator();
-		if(generator == null) {
-			log.warn("no IFlowGenerator found. Unable to generate flow diagrams");
+		String generatorBeanClass = AppConstants.getInstance().getProperty("flow.generator");
+		if(StringUtils.isNotEmpty(generatorBeanClass)) {
+			flowGenerator = createFlowGenerator(generatorBeanClass);
 		} else {
-			if(log.isDebugEnabled()) log.debug("using IFlowGenerator ["+generator+"]");
-			fileExtension = generator.getFileExtension();
+			log.info("no FlowGenerator found. Unable to generate flow diagrams");
```

Comment:
Ik vind deze melding dramatischer dan het is. 
```suggestion
			log.info("no FlowGenerator configured. No flow diagrams will be generated");
```